internal/mode/static/build_statuses.go

```diff
@@ -141,10 +141,10 @@ func buildGatewayStatus(
 		}
 	}
 
-	listenerStatuses := make(map[string]status.ListenerStatus)
+	listenerStatuses := []status.ListenerStatus{}
```

Comment:
We prefer to [construct slices with the known capacity](https://github.com/nginxinc/nginx-gateway-fabric/blob/main/docs/developer/go-style-guide.md#construct-slices-with-known-capacity).

```suggestion
	listenerStatuses := make([]string, 0, len(gateway.Listeners))
```

Reply:
Updated https://github.com/nginxinc/nginx-gateway-fabric/pull/1324/commits/4cc8cd0a9b3e3c951028bed1e6776dd7797e4861.